dora-metrics/pelorus
# Fix: Pin Python dependencies

## Linked Issues

resolves #951
depends on #948

## Description

Avoid code breaking with unexpected updates from Python dependencies.

Now, poetry will manage the Python dependencies for us. To add (or update) a dependency, run a poetry command (it will check if there are no conflicts with the current dependencies, before adding it). This will update both `pyproject.toml` and `poetry.lock`, but not requirements files. To update them, run the new make command (we will not update requirements file anymore, this is poetry job). Requirements files now have more things in there:
- the previous dependencies, but now all pinned (no more unexpected updates to break the project CI)
- the dependencies of the dependencies (that were already installed previously, but were under the rug, and we had no control over them, which poetry also monitors)
- the hash, which is a security constrain  

Dependabot will open PRs when there are updates available, and the new workflow will warn if the requirements files where not updated.

## Testing Instructions

Run documentation commands and check if nothing broke with the changes (for example `make dev-env`).

To test the built exporters images with pinned versions, run the command github-action[bot] commented on the PR.
